src/H5Tconv_float.c

```diff
@@ -688,602 +755,628 @@ H5T__conv_f_i(const H5T_t *src_p, const H5T_t *dst_p, H5T_cdata_t *cdata, const
-                                    conv_ctx->u.conv.dst_type_id, src_rev, d,
-                                    conv_ctx->u.conv.cb_struct.user_data);
-                            }
-
-                            if (except_ret == H5T_CONV_UNHANDLED) {
-                                if (H5T_SGN_2 == dst.u.i.sign)
-                                    H5T__bit_set(d, dst.prec - 1, (size_t)1, true);
-                            }
-                            else if (except_ret == H5T_CONV_HANDLED) {
-                                /*No need to reverse the order of destination because user handles it*/
-                                reverse = false;
-                                goto next;
-                            }
-                            else if (except_ret == H5T_CONV_ABORT)
-                                HGOTO_ERROR(H5E_DATATYPE, H5E_CANTCONVERT, FAIL,
-                                            "can't handle conversion exception");
-                        }
-                        else { /* +Infinity */
-                            if (conv_ctx->u.conv.cb_struct
-                                    .func) { /*If user's exception handler is present, use it*/
-                                /*reverse order first*/
-                                H5T__reverse_order(src_rev, s, src_p->shared->size,
-                                                   src_p->shared->u.atomic.order);
-                                except_ret = (conv_ctx->u.conv.cb_struct.func)(
-                                    H5T_CONV_EXCEPT_PINF, conv_ctx->u.conv.src_type_id,
-                                    conv_ctx->u.conv.dst_type_id, src_rev, d,
-                                    conv_ctx->u.conv.cb_struct.user_data);
-                            }
-
-                            if (except_ret == H5T_CONV_UNHANDLED) {
-                                if (H5T_SGN_NONE == dst.u.i.sign)
-                                    H5T__bit_set(d, dst.offset, dst.prec, true);
-                                else if (H5T_SGN_2 == dst.u.i.sign)
-                                    H5T__bit_set(d, dst.offset, dst.prec - 1, true);
-                            }
-                            else if (except_ret == H5T_CONV_HANDLED) {
-                                /*No need to reverse the order of destination because user handles it*/
-                                reverse = false;
-                                goto next;
-                            }
-                            else if (except_ret == H5T_CONV_ABORT)
-                                HGOTO_ERROR(H5E_DATATYPE, H5E_CANTCONVERT, FAIL,
-                                            "can't handle conversion exception");
-                        }
-                        goto padding;
+
+                    if (except_ret == H5T_CONV_UNHANDLED)
+                        H5T__bit_set(d, (dst_atomic.offset + dst_atomic.prec - 1), (size_t)1, true);
```

Comment:
We're using the H5T__bit_set, etc. routines so heavily now, I suggest moving them into one of the H5T headers and making them 'static inline', to speed them up.